tests/test_other.py

```diff
@@ -2425,20 +2425,59 @@ def compile_to_O0_executable(compile_args):
     self.assertEqual(no_size, line_size)
     self.assertEqual(line_size, full_size)
 
+  # Verify the existence (or lack thereof) of DWARF info in the given wasm file
+  def verify_dwarf(self, wasm_file, verify_func):
+    self.assertExists(wasm_file)
+    info = self.run_process([LLVM_DWARFDUMP, '--all', wasm_file], stdout=PIPE).stdout
+    verify_func('DW_TAG_subprogram', info) # Subprogram entry in .debug_info
+    verify_func('debug_line[0x', info) # Line table
+
+  def verify_dwarf_exists(self, wasm_file):
+    self.verify_dwarf(wasm_file, self.assertIn)
+
+  def verify_dwarf_does_not_exist(self, wasm_file):
+    self.verify_dwarf(wasm_file, self.assertNotIn)
+
+  # Verify if the given file name contains a source map
+  def verify_source_map_exists(self, map_file):
+    self.assertExists(map_file)
+    data = json.load(open(map_file))
+    # Simply check the existence of required sections
+    self.assertIn('version', data)
+    self.assertIn('sources', data)
+    self.assertIn('mappings', data)
+
   def test_dwarf(self):
     def compile_with_dwarf(args, output):
       # Test that -g enables dwarf info in object files and linked wasm
       self.run_process([EMXX, test_file('hello_world.cpp'), '-o', output, '-g'] + args)
-
-    def verify(output):
-      info = self.run_process([LLVM_DWARFDUMP, '--all', output], stdout=PIPE).stdout
-      self.assertIn('DW_TAG_subprogram', info) # Ensure there's a subprogram entry in .debug_info
-      self.assertIn('debug_line[0x', info) # Ensure there's a line table
-
     compile_with_dwarf(['-c'], 'a.o')
-    verify('a.o')
+    self.verify_dwarf_exists('a.o')
     compile_with_dwarf([], 'a.js')
-    verify('a.wasm')
+    self.verify_dwarf_exists('a.wasm')
+
+  def test_dwarf_with_source_map(self):
+    source_file = 'hello_world.cpp'
+    js_file = 'a.out.js'
+    wasm_file = 'a.out.wasm'
+    map_file = 'a.out.wasm.map'
+
+    # Generate only DWARF
+    self.emcc(test_file(source_file), ['-g'], js_file)
+    self.verify_dwarf_exists(wasm_file)
+    self.assertFalse(os.path.isfile(map_file))
+    try_delete([wasm_file, map_file, js_file])
```

Comment:
Actually I think you can just `self.clear()` here instead and avoid the new helper completely?

Reply:
Oh, didn't know that exists. But I guess the new helper can be useful in some places?

Reply:
Used `self.clear()` in #17519.

Reply:
I make it part of a bigger refactoring in https://github.com/emscripten-core/emscripten/pull/17512.. thanks for reminding about that code I've been meaning to clean up :)